brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java

```diff
@@ -127,6 +128,8 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
             ctxExtension.hook(() -> currentTraceContext.newScope(span.context()));
         }
 
+        ctx.logBuilder().tracingContext(SpanConverter.toLogTracingContext(span));
```

Comment:
@minwoox @KonaEspresso94 if this is only for access logs, my main question here is that is this a push/pop sort of thing. so for example, is this masking the server request log? or is it a different object?

Note: the code above this pushes the span context already, just to integrate with brave instead of the log builder.
```

        final RequestContextExtension ctxExtension = ctx.as(RequestContextExtension.class);
        if (currentTraceContext != null && !span.isNoop() && ctxExtension != null) {
            // Make the span the current span and run scope decorators when the ctx is pushed.
            ctxExtension.hook(() -> currentTraceContext.newScope(span.context()));
        }
```

where you see the "hook" part seems like the place to do something like this, as it only triggers where there is an active span (not no-op due to disabled or otherwise). So if we go with this change, I think we do it inside the same if.

second part: push something derived from the span context not the span, as the context is ultimately what has the properties needed.

third part: If we want to push the context for retrieval of IDs, basically sounds fine, just we need to document what value someone should expect and have unit tests for client spans and server spans (that the IDs are what we think they should be). The documentation should say that the TraceContext (really should be that not TracingContext but that's a nit) is only present when there is a sampled span.

If those IDs are the "current client or server span" then copying off the values works fine, because this logic inside brave it will be cached, so not create additional garbage.

```java
  volatile String spanIdString; // Lazily initialized and cached.

  /** Returns the hex representation of the span's ID */
  public String spanIdString() {
    String r = spanIdString;
    if (r == null) {
      r = spanIdString = toLowerHex(spanId);
    }
    return r;
  }
```

So, here's the summary advice for implementing this for AccessLog:

1. decide whether `TracingContext (adrian likes TraceContext better) ` should be a class or an interface. probably if we want to guarantee the value doesn't change, class is best despite the cost of populating it.
2. document the `TracingContext` holds the last client or server span ID, and is only present when when there is a sampled span (so null when that's not the case for any reason)
3. move the populating part to inside the same if statement as the ctxExtension.hook, so no overhead unless there's a real context
4. add integration tests for client and server together.
5. showcase how this is actually used (most important part) as there's already more flexible log integration with SLFJ and brave.